demo/loadmore.html

```diff
@@ -0,0 +1,83 @@
+<html>
+
+<head>
+  <title>Load more</title>
+  <meta name="viewport" content="width=device-width, minimum-scale=1.0, initial-scale=1, user-scalable=yes">
+
+  <style>
+    body {
+      font-family: Roboto, Helvetica, sans-serif;
+      max-width: 600px;
+      height: 100vh;
+      margin: 0px auto;
+      display: flex;
+      flex-direction: column;
+    }
+
+    virtual-scroller {
+      flex: 1;
+      margin: 10px 0;
+      border-top: 2px solid #eee;
+      border-bottom: 2px solid #eee;
+    }
+
+    contact-element,
+    button {
+      height: 80px;
+    }
+  </style>
+</head>
+
+<body>
+  <div id=errors></div>
+  <div>
+    <button id=swapButton>switch virtual-scroller &lt;-&gt; div</button>
+  </div>
+
+  <virtual-scroller id=scroller></virtual-scroller>
+
+  <script type="module">
+    import * as confirmFeatures from './util/confirm-features.mjs';
+    confirmFeatures.confirm(errors);
+  </script>
+  <script type="module">
+    import 'std:elements/virtual-scroller';
+    import './contacts/contact-element.mjs';
+    import { ContactDataSource } from './contacts/contact-data-source.mjs';
+    import * as util from './util/util.mjs';
+
+    const dataSource = new ContactDataSource();
+
+    const button = document.createElement('button');
+    scroller.appendChild(button);
+
+    const loadMore = async () => {
+      // Start loading the contacts...
+      const promise = dataSource.getContacts(20);
+      button.textContent = dataSource.loading ? 'Loading...' : 'Load more';
+      // ... and wait for the contacts to be loaded.
+      const newContacts = await promise;
+      button.textContent = dataSource.loading ? 'Loading...' : 'Load more';
+      let i = 0;
+      for (const contact of newContacts) {
+        let contactElement = document.createElement('contact-element');
+        scroller.insertBefore(contactElement, button);
+        contactElement.contact = contact;
+        contactElement.id = "c" + i;
```

Comment:
Are the IDs necessary?

Reply:
No and they're not correct either since each batch reuses 1..20. Deleted.